python/packages/ollama/agent_framework_ollama/_chat_client.py

```diff
@@ -0,0 +1,312 @@
+                " TextReasoningContent, or FunctionCallContent."
+            )
+
+        assistant_message = OllamaMessage(role="assistant", content=message.text, thinking=message.reasoning)
+
+        tool_calls = [item for item in message.contents if isinstance(item, FunctionCallContent)]
+        if tool_calls:
+            assistant_message["tool_calls"] = [
+                {
+                    "function": {
+                        "call_id": tool_call.call_id,
+                        "name": tool_call.name,
+                        "arguments": tool_call.arguments
+                        if isinstance(tool_call.arguments, Mapping)
+                        else json.loads(tool_call.arguments or "{}"),
+                    }
+                }
+                for tool_call in tool_calls
+            ]
+        return [assistant_message]
+
+    def _format_tool_message(self, message: ChatMessage) -> list[OllamaMessage]:
+        # Ollama does not support multiple tool results in a single message, so we create a separate
+        function_result_items = [
+            OllamaMessage(role="tool", content=str(item.result), tool_name=item.call_id)
+            for item in message.contents
+            if isinstance(item, FunctionResultContent)
+        ]
+
+        if not function_result_items:
+            raise ValueError("Tool message must have a function result content item.")
+
+        return function_result_items
+
+    def _ollama_to_agent_framework_content(self, response: OllamaChatResponse) -> list[Contents]:
+        contents: list[Contents] = []
+        if response.message.thinking:
+            contents.append(TextReasoningContent(text=response.message.thinking))
+        if response.message.content:
+            contents.append(TextContent(text=response.message.content))
+        if response.message.tool_calls:
+            tool_calls = self._parse_ollama_tool_calls(response.message.tool_calls)
+            contents.extend(tool_calls)
+        return contents
+
+    def _ollama_streaming_response_to_agent_framework_message(self, response: OllamaChatResponse) -> ChatResponseUpdate:
+        contents = self._ollama_to_agent_framework_content(response)
+        return ChatResponseUpdate(
+            contents=contents,
+            role=Role.ASSISTANT,
+            ai_model_id=response.model,
+            created_at=response.created_at,
+        )
+
+    def _ollama_response_to_agent_framework_message(self, response: OllamaChatResponse) -> ChatResponse:
+        contents = self._ollama_to_agent_framework_content(response)
+
+        return ChatResponse(
+            messages=[ChatMessage(role=Role.ASSISTANT, contents=contents)],
+            model_id=response.model,
+            created_at=response.created_at,
+            usage_details=UsageDetails(
+                input_token_count=response.prompt_eval_count,
+                output_token_count=response.eval_count,
+                total_token_count=(
```

Comment:
this doesn't add much value, if it not passed by the client. Total is only for api's that just return that and no details.